packages/osd-stylelint-plugin-stylelint/src/utils/compliance_engine.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Any modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+export interface ComplianceRule {
```

Comment:
Can we have some documenting on this compliance engine, as in how to introduce new rule, how to tweak existing rule etc.

Reply:
Agreed added a fast follow: https://github.com/opensearch-project/oui/issues/846